compiler/src/dotty/tools/dotc/transform/CheckUnused.scala

```diff
@@ -674,47 +696,44 @@ object CheckUnused:
     extension (sym: Symbol)
       /** is accessible without import in current context */
       private def isAccessibleAsIdent(using Context): Boolean =
-        sym.exists &&
-          ctx.outersIterator.exists{ c =>
-            c.owner == sym.owner
-            || sym.owner.isClass && c.owner.isClass
-                && c.owner.thisType.baseClasses.contains(sym.owner)
-                && c.owner.thisType.member(sym.name).alternatives.contains(sym)
-          }
+        ctx.outersIterator.exists{ c =>
+          c.owner == sym.owner
+          || sym.owner.isClass && c.owner.isClass
+              && c.owner.thisType.baseClasses.contains(sym.owner)
+              && c.owner.thisType.member(sym.name).alternatives.contains(sym)
+        }
 
       /** Given an import and accessibility, return selector that matches import<->symbol */
-      private def isInImport(imp: tpd.Import, isAccessible: Boolean, altName: Option[Name], isDerived: Boolean)(using Context): Option[ImportSelector] =
-        val tpd.Import(qual, sels) = imp
-        val qualTpe = qual.tpe
-        val dealiasedSym = sym.dealias
-        val simpleSelections = qualTpe.member(sym.name).alternatives
-        val selectionsToDealias = sels.flatMap(sel =>
-          qualTpe.member(sel.name.toTypeName).alternatives
-          ::: qualTpe.member(sel.name.toTermName).alternatives)
-        def qualHasSymbol = simpleSelections.map(_.symbol).contains(sym) || (simpleSelections ::: selectionsToDealias).map(_.symbol).map(_.dealias).contains(dealiasedSym)
-        def selector = sels.find(sel => (sel.name.toTermName == sym.name || sel.name.toTypeName == sym.name) && altName.map(n => n.toTermName == sel.rename).getOrElse(true))
-        def dealiasedSelector =
-          if isDerived then
-            sels.flatMap(sel => selectionsToDealias.map(m => (sel, m.symbol))).collect {
-              case (sel, sym) if sym.dealias == dealiasedSym => sel
-            }.headOption
-          else None
-        def givenSelector = if sym.is(Given) || sym.is(Implicit)
-          then sels.filter(sel => sel.isGiven && !sel.bound.isEmpty).find(sel => sel.boundTpe =:= sym.info)
-          else None
-        def wildcard = sels.find(sel => sel.isWildcard && ((sym.is(Given) == sel.isGiven && sel.bound.isEmpty) || sym.is(Implicit)))
-        if sym.exists && qualHasSymbol && (!isAccessible || sym.isRenamedSymbol(altName)) then
-          selector.orElse(dealiasedSelector).orElse(givenSelector).orElse(wildcard) // selector with name or wildcard (or given)
+      private def isInImport(selData: ImportSelectorData, altName: Option[Name], isDerived: Boolean)(using Context): Boolean =
+        assert(sym.exists)
+
+        val selector = selData.selector
+
+        if !selector.isWildcard then
+          if !altName.forall(explicitName => selector.rename == explicitName.toTermName) then
```

Comment:
Can this be changed to `altName.exists(explicitName => selector.rename != explicitName.toTermName)`? (It would read like the comment then)